sim/core/spell_mod.go

```diff
@@ -106,6 +117,18 @@ func shouldApply(spell *Spell, mod *SpellMod) bool {
 		return false
 	}
 
+	if mod.ResourceType > 0 && spell.Cost != nil {
```

Comment:
Probably cleaner to also return `false` if `(mod.ResourceType > 0) && (spell.Cost == nil)`.